Main.java

```diff
@@ -4,118 +4,270 @@
 
-class Trapezoid extends Quadilateral{
-  private float side1;
-  private float side2;
-  private float side3;
-  private float side4;
-  public Trapezoid (float a, float b, float c, float d){
-    side1 = a;
-    side2 = b;
-    side3 = c;
-    side4 = d;
+      return 4*ts;
+    }
+
+    public boolean check(){
+
+      return ( (ts<=0) || (d1<=0) || (d2<=0) );
+    }
+}
+
+//Create class for square
+class Square extends Quadilateral{
+
+  public Square(float a){
+    ts=a;
+  }
+
+  public float Area(){
+    return ts*ts;
   }
 
   public float calculatePerimeter(){
-    return side1+side2+side3+side4;
+    return 4*ts;
   }
+  
+}
+
+
+//create class for Rectangle
+class Rectangle extends Quadilateral{
+
+    public Rectangle(float a,float b){
+      ts=a;
+      ls=b;
+    }
+
+    public float Area(){
+      return ts*ls;
+    }
+
+    public float calculatePerimeter(){
+      return (ts+ls)*2;
+    }
+
+}
+
+class Trapezoid extends Quadilateral{
+
+    public Trapezoid (float a, float b, float c, float d,float h){
+      ts = a;
+      rs = b;
+      bs = c;
+      ls = d;
+      height=h;
+    }
+
+    public float calculatePerimeter(){
+      return ts+rs+bs+ls;
+    }
+
+    public float Area(){
+      return ((ts+bs)/2)*height;
+
+    }
+
 }
 class Parallelogram extends Quadilateral{
-  private float base;
-  private float height;
-  public Parallelogram (float b, float h){
-    if (b <= 0){
-      System.out.println("You cannot have a negative or zero base. Therfore base returns to default which is 1");
-      base = 1;
-    }
-    else
-      base = b;
-    if (h <= 0){
-      System.out.println("You cannot have a negative or zero height. Therfore height returns to default which is 1");
-      height = 1;
-    }
-    else
-      height = h;
-  }
-  public float Area(){
-    return base*height;
-  }
+
+    public Parallelogram (float l,float b, float h){
+      height=h;
+      bs=b;
+      ls=l;
+    }
+
+    public float Area(){
+      return height*bs;
+    }
+
+    public float calculatePerimeter(){
+      return (bs+ls)*2;
+    }
 
 }
-class Quadilateral {
-  private float length;
-  private float breadth;
 
-  public void setLength(float l){
-    length = l;
-  }
+ class Quadilateral {
 
-  public float getLength(){
-    return length;
-  }
+    // ts top side
+    // bs bottom side
+    // ls left side
+    // rs right side
+    protected float ts=1;
+    protected float bs=1;
+    protected float ls=1;
+    protected float rs=1;
+    protected float height=1f;
 
-  public void setBreadth(float b){
-    breadth = b;
-  }
+    public float calculatePerimeter(){
+
+      return 2*(ls+bs);
+    }
+
+    public float Area(){
+      return bs*ls;
+    }
+    
+    public boolean check(){
+
+      return ( (ts<=0f) || (bs<=0f) || (rs<=0f) || (ls<=0f) );
+    }
 
-  public float getBreadth(){
-    return breadth;
-  }
 
-  public float calculatePerimeter(){
-    return 2*(length+breadth);
-  }
 }
```

Comment:
Deducted four marks from your scores as it is obvious you copied Tolulope Fakoya